conda_forge_tick/update_upstream_versions.py

```diff
@@ -392,55 +434,68 @@ def _update_upstream_versions_sequential(gx, sources):
         with node_attrs as attrs:
             try:
                 new_version = get_latest_version(attrs, sources)
-                attrs["new_version"] = new_version or attrs['new_version']
+                attrs["new_version"] = new_version or attrs["new_version"]
             except Exception as e:
                 try:
                     se = str(e)
                 except Exception as ee:
-                    se = "Bad exception string: {}".format(ee)
-                logger.warn("Error getting uptream version of {}: {}".format(node, se))
+                    se = f"Bad exception string: {ee}"
+                logger.warning(f"Error getting uptream version of {node}: {se}")
                 attrs["bad"] = "Upstream: Error getting upstream version"
             else:
                 logger.info(
-                    "{} - {} - {}".format(node, attrs.get("version"), attrs.get("new_version"))
+                    "{} - {} - {}".format(
+                        node, attrs.get("version"), attrs.get("new_version"),
+                    ),
                 )
 
 
-def _update_upstream_versions_process_pool(gx, sources):
+def _update_upstream_versions_process_pool(
+    gx: nx.DiGraph, sources: Iterable[AbstractSource],
+) -> None:
     futures = {}
-    with executor(kind='dask', max_workers=20) as (pool, as_completed):
+    with executor(kind="dask", max_workers=20) as pool:
         for node, node_attrs in gx.nodes.items():
-            with node_attrs['payload'] as attrs:
+            with node_attrs["payload"] as attrs:
                 if attrs.get("bad") or attrs.get("archived"):
                     attrs["new_version"] = False
                     continue
                 futures.update(
-                    {pool.submit(get_latest_version, attrs, sources): (node, attrs)}
+                    {pool.submit(get_latest_version, attrs, sources): (node, attrs)},
                 )
         for f in as_completed(futures):
             node, node_attrs = futures[f]
             with node_attrs as attrs:
```

Comment:
I'm not certain if this is ok, when we use the payload one before

Reply:
this is certainly not a change though.

Reply:
Fair